docs - update publishing details, add menuTitle instructions (#37441)
grafana/grafana: docs/README.md

The side menu is automatically build from the file structure. Use the [weight](https://gohugo.io/templates/lists/#by-weight) front matter parameter to order pages.

> Use the front matter param `menuTitle` to specify different menu text from the page title.

Reviewer: ```suggestion
To specify different menu text from the page title, use the front matter parameter `menuTitle`.
```

Reviewer: Nit: Put the _what_ before _how_.

## Deploy changes to grafana.com

When a PR is merged with changes in the `docs/sources` directory, those changes are automatically synced by a GitHub action (`.github/workflows/publish.yml`) to the grafana/website repo.

> * A PR targeting the `main` branch will sync to the `content/docs/grafana/next` directory in the website repo, and publish to https://grafana.com/docs/grafana/next/.

Reviewer: ```suggestion
* A PR that targets the `main` branch syncs to the `content/docs/grafana/next` directory in the `website` repository, and publishes to `https://grafana.com/docs/grafana/next/`.
```

> * A PR targeting the latest/current release branch will sync to the `content/docs/grafana/latest` directory in the website repo, and publish to https://grafana.com/docs/grafana/latest/.

Reviewer: ```suggestion
* A PR targeting the `latest/current` release branch syncs to the `content/docs/grafana/latest` directory in the `website` repository, and publishes to `https://grafana.com/docs/grafana/latest/`.
```

> Once the sync is complete, the website will automatically publish to production - no further action is needed.

Reviewer: ```suggestion
After the sync is complete, the website will automatically publish to production.
```
What does "the website will automatically publish to production" mean? Where is that happens?
